lib/services/trustedcluster.go

```diff
@@ -43,5 +47,98 @@ func ValidateTrustedCluster(tc TrustedCluster) error {
 		})
 	}
 
+	if err := ValidateRoleMap(tc.GetRoleMap()); err != nil {
+		return trace.Wrap(err)
+	}
+
 	return nil
 }
+
+// ValidateRoleMap checks RoleMap for errors
+func ValidateRoleMap(r RoleMap) error {
+	_, err := parseRoleMap(r)
+	return trace.Wrap(err)
+}
+
+// RoleMapToString prints user friendly representation of role mapping
```

Comment:
I don't think this is being called by anyone? You can drop it then move body of `parseRoleMap` into `ValidateRoleMap`.